op-challenger/fault/solver_test.go

```diff
@@ -0,0 +1,107 @@
+	"github.com/stretchr/testify/require"
+)
+
+// TestSolver_NextMove_InvalidOpponentValidParent tests the [Solver] NextMove
+// function with an invalid opponent [Claim] and valid parent [Claim].
+func TestSolver_NextMove_InvalidOpponentValidParent(t *testing.T) {
+	tests := []struct {
+		claim  Claim
+		parent Claim
+		res    *Response
+	}{
+		{
+			claim: Claim{
+				Value:    common.HexToHash("0x000000000000000000000000000000000000000000000000000000000000077a"),
+				Position: NewPosition(0, 0),
+			},
+			parent: Claim{
+				Value:    common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000768"),
+				Position: NewPosition(0, 0),
+			},
+			res: &Response{
+				Attack: true,
+				Value:  common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000364"),
+			},
+		},
+	}
+
+	gameDepth := 3
+	canonical := "abcdefgh"
+	canonicalProvider := NewAlphabetProvider(canonical, uint64(gameDepth))
+	solver := NewSolver(gameDepth, canonicalProvider)
+
+	for _, test := range tests {
+		res, err := solver.NextMove(test.claim, test.parent)
+		require.NoError(t, err)
+		require.Equal(t, test.res, res)
+	}
+}
+
+// TestSolver_NextMove_Opponent tests the [Solver] NextMove function
+// with an [fault.AlphabetProvider] as the [TraceProvider].
+func TestSolver_NextMove_Opponent(t *testing.T) {
+	// Construct the solver.
+	maxDepth := 3
+	canonicalProvider := NewAlphabetProvider("abcdefgh", uint64(maxDepth))
+	solver := NewSolver(maxDepth, canonicalProvider)
+
+	// Construct the root parent claim.
+	parentClaim := Claim{
+		Value:    common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000768"),
+		Position: NewPosition(0, 0),
+	}
+
+	// The following claims are created using the state: "abcdexyz".
+	// The responses are the responses we expect from the solver.
+	indices := []struct {
+		traceIndex int
+		claim      Claim
+		response   *Response
+	}{
+		{
+			7,
+			Claim{
+				Value:    common.HexToHash("0x000000000000000000000000000000000000000000000000000000000000077a"),
+				Position: NewPosition(0, 0),
+			},
+			&Response{
+				Attack: true,
+				Value:  common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000364"),
+			},
+		},
+		{
+			3,
+			Claim{
+				Value:    common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000364"),
+				Position: NewPosition(1, 0),
+			},
+			&Response{
+				Attack: false,
+				Value:  common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000566"),
+			},
+		},
+		{
+			5,
+			Claim{
+				Value:    common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000578"),
+				Position: NewPosition(2, 2),
+			},
+			&Response{
+				Attack: true,
+				Value:  common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000465"),
+			},
+		},
+	}
+
+	for _, test := range indices {
+		res, err := solver.NextMove(test.claim, parentClaim)
```

Comment:
Parent Claim should not be static - it should be layer higher than the claim. It's probably fine to skip the first test case because that's the root claim which is weird.

Reply:
It isn't static since the current position API mutates this variable inplace through the `Attack` and `Defend` methods.

Reply:
Oops im incorrect, changing.